src/core/include/openvino/core/type.hpp

```diff
@@ -72,6 +86,7 @@ struct OPENVINO_API DiscreteTypeInfo {
     size_t hash();
 
 private:
+    mutable std::mutex _mutex;
```

Comment:
it's too heavy solution..

Reply:
does this data race really turn into issues? Overwrite size_t does not seem to be a real problem (it's not a container).